hazelcast/src/main/java/com/hazelcast/executor/impl/DistributedExecutorService.java

```diff
@@ -298,6 +299,9 @@ private Processor(String name, UUID uuid,
 
         @Override
         public void run() {
+            String currentThreadName = Thread.currentThread().getName();
```

Comment:
The change seems correct, I wonder if we could make it more broad though, ie. for the other Executor services to take advantage of it. For example, ScheduledExecutor & DurableExecutor both use the same core logic, delegating to Cached Executors from `ExecutionServiceImpl` thus, they should have the same issue as IExecutorService.
Could this be logic in the ThreadFactory in there, or too complicated?